mlrun/package/packagers/default_packager.py

```diff
@@ -0,0 +1,594 @@
+      pack + unpack class methods implemented to collect the supported artifact types. If ``PackagerX`` has ``pack_y``,
+      ``unpack_y`` and ``pack_z``, ``unpack_z`` that means the artifact types supported are 'y' and 'z'.
+    * The abstract class method ``get_default_artifact_type``: The method is implemented to return the new class
+      variable ``DEFAULT_ARTIFACT_TYPE``. You may still override the method if the default artifact type you need may
+      change according to the object that's about to be packaged.
+
+    Important to remember (from the ``Packager`` docstring):
+
+    * Linking artifacts ("extra data"): In order to link between packages (using the extra data or metrics spec
+      attributes of an artifact), you should use the key as if it exists and as value ellipses (...). The manager will
+      link all packages once it is done packing.
+
+      For example, given extra data keys in the log hint as `extra_data`, setting them to an artifact should be::
+
+          artifact = Artifact(key="my_artifact")
+          artifact.spec.extra_data = {key: ... for key in extra_data}
+    * Clearing outputs: Some packagers may produce files and temporary directories that should be deleted once done with
+      logging the artifact. The packager can mark paths of files and directories to delete after logging using the class
+      method ``future_clear``.
+
+      For example, in the following packager's ``pack`` method we can write a text file, create an Artifact and then
+      mark the text file to be deleted once the artifact is logged::
+
+          with open("./some_file.txt", "w") as file:
+              file.write("Pack me")
+          artifact = Artifact(key="my_artifact")
+          cls.future_clear(path="./some_file.txt")
+          return artifact, None
+    """
+
+    # The type of object this packager can pack and unpack:
+    PACKABLE_OBJECT_TYPE: Type = ...
+    # The default artifact type to pack as or unpack from:
+    DEFAULT_ARTIFACT_TYPE = ArtifactType.OBJECT
+
+    @classmethod
+    def get_default_artifact_type(cls, obj: Any) -> str:
+        """
+        Get the default artifact type of this packager.
+
+        :param obj: The about to be packed object.
+
+        :return: The default artifact type.
+        """
+        return cls.DEFAULT_ARTIFACT_TYPE
+
+    @classmethod
+    def get_supported_artifact_types(cls) -> List[str]:
+        """
+        Get all the supported artifact types on this packager.
+
+        :return: A list of all the supported artifact types.
+        """
+        # We look for pack + unpack method couples so there won't be a scenario where an object can be packed but not
+        # unpacked. Result has no unpacking so we add it separately.
+        return [
+            key[len("pack_") :]
+            for key in dir(cls)
+            if key.startswith("pack_") and f"unpack_{key[len('pack_'):]}" in dir(cls)
+        ] + ["result"]
```

Comment:
Currently in MLRun every class that is returned (regardless the decorator) is logged as result. You'll see stuff like <module.ClassName> x0AFF435 in the run's results as this is the representation of the class. I keep this behavior by adding the same `str(obj)` as a default when using `artifact_type="result"`. Notice it is only set in the `DefaultPackager`, if a user would like to not have this option for whatever reason, he can override it or inherit from `Packager`.

Regarding the not have unpacking, it is a result, meaning it can be passed as a parameter, not input. So there will be nothing to unpack. It's the only case of this scenario.